test/ILP/default-ilp-client-test.ts

```diff
@@ -63,17 +56,72 @@ describe('Default ILP Client', function(): void {
     })
   })
 
+  it('Get balance - Not Found Error', function(done): void {
+    // GIVEN a DefaultIlpClient with a network client which always throws a NOT_FOUND error
+    const client = FakeDefaultIlpClients.fakeNotFoundErrorClient()
+
+    // WHEN the balance for an account is requested
+    client.getBalance('test.foo.bar').catch((error) => {
+      // THEN the error is translated to a XpringIlpError
+      assert.equal(error as XpringIlpError, XpringIlpError.accountNotFound)
+      done()
+    })
+  })
+
+  it('Get balance - Invalid Argument Error', function(done): void {
+    // GIVEN a DefaultIlpClient with a network client which always throws an INVALID_ARGUMENT error
+    const client = FakeDefaultIlpClients.fakeInvalidArgumentErrorClient()
+
+    // WHEN the balance for an account is requested
+    client.getBalance('test.foo.bar').catch((error) => {
+      // THEN the error is translated to a XpringIlpError
+      assert.equal(error as XpringIlpError, XpringIlpError.invalidArgument)
+      done()
+    })
+  })
+
+  it('Get balance - Unauthenticated Error', function(done): void {
+    // GIVEN a DefaultIlpClient with a network client which always throws an UNAUTHENTICATED error
+    const client = FakeDefaultIlpClients.fakeUnauthenticatedErrorClient()
+
+    // WHEN the balance for an account is requested
+    client.getBalance('test.foo.bar').catch((error) => {
+      // THEN the error is translated to a XpringIlpError
+      assert.equal(error as XpringIlpError, XpringIlpError.unauthenticated)
+      done()
+    })
+  })
+
+  it('Get balance - Internal Error', function(done): void {
+    // GIVEN a DefaultIlpClient with a network client which always throws an INTERNAL error
+    const client = FakeDefaultIlpClients.fakeInternalErrorClient()
+
+    // WHEN the balance for an account is requested
+    client.getBalance('test.foo.bar').catch((error) => {
+      // THEN the error is translated to a XpringIlpError
+      assert.equal(error as XpringIlpError, XpringIlpError.internal)
+      done()
+    })
+  })
+
+  it('Get balance - Invalid Access Token', function(done): void {
+    // GIVEN a DefaultIlpClient with a network client which always throws a XpringIlpError.invalidAccessToken error
+    const client = FakeDefaultIlpClients.fakeInvalidAccessTokenErrorClient()
+
+    // WHEN the balance for an account is requested
+    client.getBalance('test.foo.bar').catch((error) => {
+      // THEN the error is rethrown
```

Comment:
I think this is just being translated to a `XpringIlpError`?

Reply:
oh no I see this one is already a `XpringIlpError` to begin with

Reply:
Yep. The network client throws a `XpringIlpError` if the access token is invalid, and doesn't actually make a network call, but `DefaultIlpClient` needs to handle both RPC errors and `XpringIlpErrors`